modules/mapping-utils/src/resolveSets.js

```diff
@@ -149,48 +175,193 @@ export const deleteSaveSets = ({ callback }) => async (
   return esResponse.deleted;
 };
 
-export const renameSaveSetTag = ({ callback }) => async (
+export const updateSet = ({ types, callback }) => async (
   obj,
-  { setId, tag, userId },
+  { source, subAction, target, userId, data },
   { es },
 ) => {
-  const esResponse = await es.updateByQuery({
-    index: CONSTANTS.ES_ARRANGER_SET_INDEX,
-    type: CONSTANTS.ES_ARRANGER_SET_TYPE,
-    refresh: true,
-    body: {
-      script: {
-        lang: 'painless',
-        source: `ctx._source['tag'] = '${tag}'`,
-      },
-      query: {
-        bool: {
-          filter: {
-            term: { userId: userId },
+  const { sourceType } = source;
+
+  switch (subAction) {
+    case SubActionTypes.REMOVE_IDS:
+    case SubActionTypes.ADD_IDS: {
+      if (SourceType.QUERY === sourceType) {
+        const { type, sqon, path } = data;
+
+        if (isQueryEmpty(sqon)) {
```

Comment:
I don't see how a empty query is a exception. This a a completly valid state, maybe we can't work on it and need to do something, but still a valid state, it's not 'exceptional'

maybe there is something I don't yet understand with arranger. But throwing here and there for business logic a la java, is IMO, a bad pratice. We should be clear about the flow.

Plus, the exception seem to be unhandled later on.

https://dzone.com/articles/exceptions-as-controlflow-in-java#:~:text=One%20of%20these%20common%20bad,makes%20your%20code%20less%20readable.&text=For%20this%20reason%2C%20the%20functional%20programming%20paradigm%20does%20not%20welcome%20exceptions.

https://reflectoring.io/business-exceptions/

Reply:
Absolutely. Thanks